sdk/metric/sdk.go

```diff
@@ -216,7 +216,13 @@ func (a *asyncInstrument) getRecorder(kvs []core.KeyValue) export.Aggregator {
 
 	lrec, ok := a.recorders[labels.ordered]
 	if ok {
-		lrec.modifiedEpoch = a.meter.currentEpoch
+		if lrec.modifiedEpoch == a.meter.currentEpoch {
+			// last value wins for Observers, so if we see the same labels
+			// in the current epoch, we replace the old recorder
+			lrec.recorder = a.meter.batcher.AggregatorFor(&a.descriptor)
```

Comment:
I think I like this approach because it uses no extra memory unless the caller produces multiple observations.

Reply:
(Not sure why I typed "I think" above. I like this approach.)